src/pages/resultsView/enrichments/AlterationEnrichmentsContainer.tsx

```diff
@@ -145,49 +124,64 @@ export default class AlterationEnrichmentContainer extends React.Component<IAlte
         }
     );
 
-    @computed get columns() {
+    @computed get customColumns() {
+        return getGroupColumns(this.props.groups, this.props.alteredVsUnalteredMode);
+    }
+
+    @computed get visibleOrderedColumnNames() {
         const columns = [];
         columns.push(AlterationEnrichmentTableColumnType.GENE,
             AlterationEnrichmentTableColumnType.CYTOBAND);
         if (this.props.showCNAInTable) {
             columns.push(AlterationEnrichmentTableColumnType.ALTERATION);
         }
-        columns.push(AlterationEnrichmentTableColumnType.PERCENTAGE_IN_GROUP1,
-        AlterationEnrichmentTableColumnType.PERCENTAGE_IN_GROUP2,
-        AlterationEnrichmentTableColumnType.LOG_RATIO,
+        this.props.groups.forEach(group=>{
+            columns.push(group.name);
+        })
+        if(this.isTwoGroupAnalysesPossible) {
+            columns.push("Log Ratio");
+        }
+        
+        columns.push(
         AlterationEnrichmentTableColumnType.P_VALUE,
-        AlterationEnrichmentTableColumnType.Q_VALUE,
-        AlterationEnrichmentTableColumnType.TENDENCY);
+        AlterationEnrichmentTableColumnType.Q_VALUE);
+
+        if(this.isTwoGroupAnalysesPossible) {
+            columns.push(this.props.alteredVsUnalteredMode ? "Tendency" : "Enriched in");
```

Comment:
you cant have "Enriched in" for more than 2 groups?

Reply:
also we should definitely use an enum for this

Reply:
yes, its value is based on LogRatio which won't be calculated if there are more than 2 group

Reply:
can these columsn be added to AlterationEnrichmentTableColumnType?